enterprise_subsidy/apps/fulfillment/tests/test_api.py

```diff
@@ -0,0 +1,202 @@
+        """
+        Ensure basic happy path of `can_fulfill` for Exec Ed (can fulfill)
+        """
+        mock_get_content_summary.return_value = {
+            'content_uuid': 'course-v1:edX-test-course',
+            'content_key': 'course-v1:edX-test-course',
+            'source': 'edX',
+            'mode': 'verified',
+            'content_price': 10000,
+            'geag_variant_id': uuid4(),
+        }
+        transaction = TransactionFactory.create(
+            state=TransactionStateChoices.PENDING,
+            quantity=-19998,
+            ledger=self.subsidy.ledger,
+            lms_user_id=self.lms_user_id,
+            content_key=self.content_key
+        )
+        assert self.geag_fulfillment_handler.can_fulfill(transaction)
+
+    def test_save_reference(self):
+        """
+        Ensure basic happy path of `_save_fulfillment_reference`
+        """
+        transaction = TransactionFactory.create(
+            state=TransactionStateChoices.PENDING,
+            quantity=-19998,
+            ledger=self.subsidy.ledger,
+            lms_user_id=self.lms_user_id,
+            content_key=self.content_key
+        )
+        external_reference_id = uuid4()
+        # pylint: disable=protected-access
+        external_ransaction_reference = self.geag_fulfillment_handler._save_fulfillment_reference(
+            transaction,
+            external_reference_id
+        )
+        assert external_ransaction_reference
+        assert external_ransaction_reference.external_reference_id == external_reference_id
+
+    @mock.patch("enterprise_subsidy.apps.content_metadata.api.ContentMetadataApi.get_content_summary")
+    def test_create_allocation_payload(self, mock_get_content_summary):
+        """
+        Ensure basic happy path of `_create_allocation_payload`
+        """
+        content_summary = {
+            'content_uuid': 'course-v1:edX-test-course',
+            'content_key': 'course-v1:edX-test-course',
+            'source': 'edX',
+            'mode': 'verified',
+            'content_price': 10000,
+            'geag_variant_id': str(uuid4()),
+        }
+        mock_get_content_summary.return_value = content_summary
+        tx_metadata = {
+            'geag_first_name': 'Donny',
+            'geag_last_name': 'Kerabatsos',
+            'geag_date_of_birth': '1900-01-01',
+            'geag_terms_accepted_at': '2021-05-21T17:32:28Z',
+        }
+        transaction = TransactionFactory.create(
+            state=TransactionStateChoices.PENDING,
+            quantity=-19998,
+            ledger=self.subsidy.ledger,
+            lms_user_id=self.lms_user_id,
+            content_key=self.content_key,
+            metadata=tx_metadata,
+        )
+        # pylint: disable=protected-access
+        geag_payload = self.geag_fulfillment_handler._create_allocation_payload(transaction)
+        assert geag_payload.get('order_items')[0].get('productId') == content_summary.get('geag_variant_id')
```

Comment:
nit: we should assert that the rest of `geag_payload` has the shape and value we expect, especially the metadata fields.